src/scipp/core/reduction.py

```diff
@@ -107,6 +107,42 @@ def nanmean(x: VariableLikeType, dim: str | None = None) -> VariableLikeType:
         return _call_cpp_func(_cpp.nanmean, x, dim=dim)  # type: ignore[return-value]
 
 
+def average(x: VariableLikeType, dim: str | None = None, weights: VariableLikeType | None = None) -> VariableLikeType:
```

Comment:
Should we have a new function or could we just add a new keyword argument to mean?

Reply:
Depends if you want parity with numpy or not.